packages/flutter/lib/src/material/tabs.dart

```diff
@@ -3,746 +3,338 @@
-      child = childParentData.nextSibling;
-    }
-    return x;
-  }
-
-  Size layoutSize;
-  List<double> layoutWidths;
-  _TabLayoutChanged onLayoutChanged;
-
-  void reportLayoutChangedIfNeeded() {
-    assert(onLayoutChanged != null);
-    List<double> widths = new List<double>(childCount);
-    if (!isScrollable && childCount > 0) {
-      double tabWidth = size.width / childCount;
-      widths.fillRange(0, widths.length, tabWidth);
-    } else if (isScrollable) {
-      RenderBox child = firstChild;
-      int childIndex = 0;
-      while (child != null) {
-        widths[childIndex++] = child.size.width;
-        final _TabBarParentData childParentData = child.parentData;
-        child = childParentData.nextSibling;
-      }
-      assert(childIndex == widths.length);
-    }
-    if (size != layoutSize || widths != layoutWidths) {
-      layoutSize = size;
-      layoutWidths = widths;
-      onLayoutChanged(layoutSize, layoutWidths);
-    }
-  }
-
-  @override
-  void performLayout() {
-    assert(constraints is BoxConstraints);
-    if (childCount == 0)
-      return;
-
-    if (isScrollable) {
-      double tabBarWidth = layoutScrollableTabs();
-      size = constraints.constrain(new Size(tabBarWidth, _tabBarHeight));
-    } else {
-      size = constraints.constrain(new Size(constraints.maxWidth, _tabBarHeight));
-      layoutFixedWidthTabs();
-    }
-
-    if (onLayoutChanged != null)
-      reportLayoutChangedIfNeeded();
-  }
-
-  @override
-  bool hitTestChildren(HitTestResult result, { Point position }) {
-    return defaultHitTestChildren(result, position: position);
-  }
-
-  void _paintIndicator(Canvas canvas, RenderBox selectedTab, Offset offset) {
-    if (indicatorColor == null)
-      return;
-
-    if (indicatorRect != null) {
-      canvas.drawRect(indicatorRect.shift(offset), new Paint()..color = indicatorColor);
-      return;
-    }
-
-    final Size size = new Size(selectedTab.size.width, _kTabIndicatorHeight);
-    final _TabBarParentData selectedTabParentData = selectedTab.parentData;
-    final Point point = new Point(
-      selectedTabParentData.offset.dx,
-      _tabBarHeight - _kTabIndicatorHeight
-    );
-    canvas.drawRect((point + offset) & size, new Paint()..color = indicatorColor);
-  }
-
-  @override
-  void paint(PaintingContext context, Offset offset) {
-    int index = 0;
-    RenderBox child = firstChild;
-    while (child != null) {
-      final _TabBarParentData childParentData = child.parentData;
-      context.paintChild(child, childParentData.offset + offset);
-      if (index++ == selectedIndex)
-        _paintIndicator(context.canvas, child, offset);
-      child = childParentData.nextSibling;
-    }
-  }
-}
-
-class _TabBarWrapper extends MultiChildRenderObjectWidget {
-  _TabBarWrapper({
+/// A material design [TabBar] tab. If both [icon] and [text] are
+/// provided, the text is displayed below the icon.
+///
+/// See also:
+///
+///  * [TabBar] - displays a row of tabs.
```

Comment:
We usually use `,` rather than `-` to separate the name from the description in the docs.